server/GameServer.java

```diff
@@ -91,13 +103,112 @@ void handleGETRequest() throws FileNotFoundException, IOException{
         fileBuffer.close();
     }
 
-    void tearDownConnection() throws IOException{
+    private void handlePOSTRequest() throws IOException{
+
+        parsePOSTRequestArgs();
+        handleUserAuthentication();
+        handleAction();
+    }
+
+    private void parsePOSTRequestArgs() throws IOException{
+
+        int length = 0;
+        String line = "";
+
+        while((line = bufferedReader.readLine()).length() != 0){
+            System.out.println(line);
+            if(line.contains("Content-Length")){
+                length = Integer.parseInt(line.split(" ")[1]);
+            }
+
+        }
+
+        char[] temp = new char[length];
+        bufferedReader.read(temp);
+        String[] kv_arr = new String(temp).split("&"); // split by key-value pair, which are separated by &;
+        for (String arg : kv_arr) {
+            String[] kv = arg.split("=");   // split by key and and value, which are separated by =
+            String key = kv[0];
+            String value = kv[1];
+            args.put(key, value);
+
+        }
+    }
+
+    private void handleAction() throws IOException{
+
+        action = args.get("action");
+        switch (action) {
+            case "user_registration":
+                // handleUserRegistration(); needs to register new user and redirect to index.html.
+                break;
+            case "login":
+                handleLogin();
+                break;
+            case "create_game":
+                // handleCreateGame();
+                break;
+            case "enter_game":
+                // handleEnterGame();
+                break;
+            case "forfeit_game":
+                // handleForfeitGame();
+                break;
+            case "move_piece":
+                // handleMovePiece();
+                break;
+            case "view_stats":
+                // handleViewStats();
+                break;
+        }
+    }
+
+    private double[] getAuthTokens(){
+
+        double username = args.get("username").hashCode() * HASH_KEY;
+        double password = args.get("password").hashCode() * HASH_KEY;
+        double pk = (username % password) * HASH_KEY;
+        return new double[]{username, password, pk};
+    }
+
+    private void handleUserAuthentication(){
+
+        String clientIP = clientSocket.getInetAddress().toString();
+        if(isLoggedIn(clientIP)){
+            return;
+        }
+
+        double[] authTokens = getAuthTokens();
+
+        // now use pk in SELECT statement to DB to lookup whether user exists.
+
+        // if user exists, bind this user to the client IP. The IP is now "logged in". Return.
+
+        // if user does not exist, return error message.
+
+    }
+
+    private boolean isLoggedIn(String clientIP){
+        // check db to see if this IP is in the "logged in" table.
+
+        boolean loggedIn = false;
+        return loggedIn;
+    }
+
```

Comment:
Is this function a placeholder? It appears to only return false.

Reply:
yep